src/routes/transactions/mappers/common/native-staking.mapper.ts

```diff
@@ -112,6 +113,47 @@ export class NativeStakingMapper {
     });
   }
 
+  /**
+   * Gets the validator public keys from logs of `deposit` transaction.
+   *
+   * @param args.transaction - the transaction object for the deposit
+   * @param args.chainId - the chain ID of the native staking deployment
+   *
+   * @returns {Array<`0x${string}`>} the validator public keys
+   */
+  private async getDepositPublicKeys(args: {
+    transaction: MultisigTransaction | ModuleTransaction | null;
+    chainId: string;
+  }): Promise<Array<`0x${string}`>> {
+    if (!args.transaction?.transactionHash) {
+      return [];
+    }
+
+    const txStatus = await this.stakingRepository.getTransactionStatus({
+      chainId: args.chainId,
+      txHash: args.transaction.transactionHash,
+    });
+
+    const depositEvents = txStatus.receipt.logs
+      .map((log) => {
+        return this.kilnDecoder.decodeDepositEvent({
+          data: log.data,
+          topics: log.topics as [
+            signature: `0x${string}`,
+            ...args: `0x${string}`[],
+          ],
+        });
+      })
+      .filter(<T>(event: T | null): event is T => event !== null);
+
+    if (depositEvents.length === 0) {
+      return [];
+    }
+
+    // There is only one DepositEvent per transaction
```

Comment:
Nit: Do you think it would be useful to emit a warning log line if this condition is not met for some reason?

Reply:
As the contracts are immutable, this should inherently never happen. However, on the off chance it does, warning makes sense. I added a `warn` and a comment in 0fe91cda00fefc23f7d6f99e1dc17df74d9d4159.